src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -48,147 +46,95 @@
-    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(myCapital, 3, player);
+    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(ProData.myCapital, 3, player);
     ProLogger.debug("Currently in defensive stance: " + isDefensive);
 
-    // Find all purchase options and min cost per hit point
-    final List<ProPurchaseOption> specialPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> factoryPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> landPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> airPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> seaPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    ProPurchaseUtils.findPurchaseOptions(player, landPurchaseOptions, airPurchaseOptions, seaPurchaseOptions,
-        factoryPurchaseOptions, specialPurchaseOptions);
-    minCostPerHitPoint = ProPurchaseUtils.getMinCostPerHitPoint(player, landPurchaseOptions);
-
     // Find the maximum number of units that can attack each territory and max enemy defenders
-    final List<Territory> myUnitTerritories =
-        Match.getMatches(data.getMap().getTerritories(), Matches.territoryHasUnitsOwnedBy(player));
-    attackOptionsUtils.findAttackOptions(player, myUnitTerritories, attackMap, unitAttackMap, transportAttackMap,
-        bombardMap, transportMapList, new ArrayList<Territory>(), new ArrayList<Territory>(),
-        new ArrayList<Territory>(), false, false);
-    attackOptionsUtils.findScrambleOptions(player, attackMap);
-    final ProMoveOptions alliedAttackOptions = attackOptionsUtils.findAlliedAttackOptions(player);
-    final ProMoveOptions enemyDefendOptions = attackOptionsUtils.findEnemyDefendOptions(player);
+    territoryManager.populateAttackOptions();
+    territoryManager.populateEnemyDefenseOptions();
 
     // Remove territories that aren't worth attacking and prioritize the remaining ones
-    final List<ProTerritory> prioritizedTerritories =
-        attackOptionsUtils.removeTerritoriesThatCantBeConquered(player, attackMap, unitAttackMap, transportAttackMap,
-            alliedAttackOptions, enemyDefendOptions, false);
-    List<Territory> territoriesToAttack = new ArrayList<Territory>();
-    for (final ProTerritory patd : prioritizedTerritories) {
-      territoriesToAttack.add(patd.getTerritory());
+    final List<ProTerritory> attackOptions = territoryManager.removeTerritoriesThatCantBeConquered();
+    List<Territory> clearedTerritories = new ArrayList<Territory>();
+    for (final ProTerritory patd : attackOptions) {
+      clearedTerritories.add(patd.getTerritory());
     }
-    ProMoveOptions enemyAttackOptions =
-        attackOptionsUtils.findEnemyAttackOptions(player, territoriesToAttack, new ArrayList<Territory>());
+    territoryManager.populateEnemyAttackOptions(clearedTerritories, new ArrayList<Territory>());
```

Comment:
This method signature should be overloaded IMO. Doesn't seem like there is a good reason that the method "populateEnemyAttackOptions" can't create it's own `new ArrayList<Territory>()`. As is it's impossible to know what that list object is being used for, if anything. Seems most likely like it's just an unused parameter.

Second, having a return type to this method would make it's functionality more clear and would also help avoid method call ordering dependencies. Consider the following where the letter a through f represent values being passed to functions:
```
foo(a, b, c);
bar(c, d);
zar(a,d); // careful, call this one after bar() only
xar(c); 
```
vs:
```
d = foo(a,b,c)
e = bar(c,d)
f = zar( a, d, e)
xar(c,f)
```
The two examples do the same thing, but one uses an input output pattern while the first calls a series of "void" methods. In the first example, it's easy to mess things up by re-ordering code. In addition, the modifications and state changes that happen are not clear, you have to follow the execution flow to know what state has changed. While with the input/output pattern, you can't screw up re-ordering the methods since the data necessary for the next call is only available from the prior calls.